websocket_subscriber/live_data_consumer.py

```diff
@@ -0,0 +1,76 @@
+import os
+import logging
+import traceback
+import threading
+import asyncio
+
+from concurrent.futures.thread import ThreadPoolExecutor
+
+from d3a_api_client.utils import logging_decorator, domain_name_from_env, \
+    consumer_websocket_domain_name_from_env
+from d3a_api_client.websocket_device import WebsocketMessageReceiver
+from d3a_api_client.constants import MAX_WORKER_THREADS
+from d3a_api_client.websocket_device import retry_coroutine
+from d3a_api_client.utils import RestCommunicationMixin
+from d3a_api_client.utils import retrieve_jwt_key_from_server
+
+
+class LiveDataWebsocketMessageReceiver(WebsocketMessageReceiver):
+    def __init__(self, rest_client):
+        super().__init__(rest_client)
+
+    def _handle_event_message(self, message):
+        if message["event"] == "live_data":
+            self.client.set_energy_forecast(**message['data'])
+        else:
+            logging.error(f"Received message with unknown event type: {message}")
+
+    def received_message(self, message):
+        try:
+            if "event" in message:
+                self._handle_event_message(message)
+        except Exception as e:
+            logging.error(f"Error while processing incoming message {message}. Exception {e}.\n"
+                          f"{traceback.format_exc()}")
+
```

Comment:
Since there are no commands nor events here, there is no need for 2 separate methods. A single `received_message` method should suffice, therefore you can delete the `_handle_event_message` and immediately call `self.client.set_energy_forecast(**message['data'])` from the `received_message` method.